### PR TITLE
Switch to using clique for merge transition simulator

### PR DESCRIPTION
**Pull request by lightclient**
This retains the ability to spawn a testnet using `ethash` for the pre-transitioned portion of the testnet, but in the configurations I've switched to using `clique` instead.

### CODE REVIEW

**Comment by fjl** [comment]
Thank you so much!

File: simulators/eth2/testnet/scenarios.go [hunk 1]
```diff
@@ -39,12 +39,12 @@ func Phase0Testnet(t *hivesim.T, env *testEnv, n node) {
 		MergeForkEpoch:          20,
 		ValidatorCount:          VALIDATOR_COUNT,
 		SlotTime:                SLOT_TIME,
-		TerminalTotalDifficulty: TERMINCAL_TOTAL_DIFFICULTY,
+		TerminalTotalDifficulty: TERMINRAL_TOTAL_DIFFICULTY,
```
Comment (holiman) [inline simulators/eth2/testnet/scenarios.go:42]:
```suggestion
		TerminalTotalDifficulty: TERMINAL_TOTAL_DIFFICULTY,
```

File: simulators/eth2/testnet/scenarios.go [hunk 2]
```diff
@@ -11,8 +11,8 @@ import (
 
 var (
 	VALIDATOR_COUNT            uint64 = 64
-	SLOT_TIME                  uint64 = 2
-	TERMINCAL_TOTAL_DIFFICULTY        = big.NewInt(5000000)
+	SLOT_TIME                  uint64 = 3
+	TERMINRAL_TOTAL_DIFFICULTY        = big.NewInt(100)
```
Comment (holiman) [inline simulators/eth2/testnet/scenarios.go:15]:
```suggestion
	TERMINAL_TOTAL_DIFFICULTY        = big.NewInt(100)
```

File: simulators/eth2/testnet/scenarios.go [hunk 3]
```diff
@@ -71,12 +71,12 @@ func TransitionTestnet(t *hivesim.T, env *testEnv, n node) {
 		MergeForkEpoch:          0,
 		ValidatorCount:          VALIDATOR_COUNT,
 		SlotTime:                SLOT_TIME,
-		TerminalTotalDifficulty: TERMINCAL_TOTAL_DIFFICULTY,
+		TerminalTotalDifficulty: TERMINRAL_TOTAL_DIFFICULTY,
```
Comment (holiman) [inline simulators/eth2/testnet/scenarios.go:74]:
```suggestion
		TerminalTotalDifficulty: TERMINAL_TOTAL_DIFFICULTY,
```

File: clients/merge-go-ethereum/Dockerfile [hunk 4]
```diff
@@ -5,8 +5,8 @@
 # needing to constantly rebuild.
 
 FROM alpine:latest
-ARG repo=MariusVanDerWijden/go-ethereum
-ARG branch=merge-kiln-v2
+ARG repo=ethereum/go-ethereum
+ARG branch=master
```
Comment (holiman) [inline clients/merge-go-ethereum/Dockerfile:9]: Can't we just remove this client now, and use regular go-ethereum?

Comment (fjl) [reply clients/merge-go-ethereum/Dockerfile:9]: Yes

File: simulators/eth2/testnet/scenarios.go (hunk 1 above)
Comment (lightclient) [reply simulators/eth2/testnet/scenarios.go:42]: Ha - sorry, went around globally replacing some things and apparently this was a casualty.

File: clients/merge-go-ethereum/Dockerfile (hunk 4 above)
Comment (lightclient) [reply clients/merge-go-ethereum/Dockerfile:9]: I've raised a separate PR #515  to remove the merge definition (based on this branch).

**Review by holiman — APPROVED** [review]
LGTM

**Pull request merged**